### ISSUE TITLE
[Archery][Integration] Consider removing `--write_generated_json` option

### ISSUE THREAD

**Issue by pitrou** [apache/arrow#37575]
### Describe the bug, including details regarding any error messages, version, and platform.

The `archery integration` has a `--write_generated_json` flag that is only used by the JS test suite:
https://github.com/apache/arrow/blob/13f0cd8ee44833efbced05c8505de12f9f02f235/js/gulp/test-task.js#L123

What the flag does is generate a bunch of JSON integration files:
https://github.com/apache/arrow/blob/13f0cd8ee44833efbced05c8505de12f9f02f235/dev/archery/archery/integration/runner.py#L502-L532

The list of generated files has not been updated in a long time. Also, the JS implementation is nowadays tested by the integration test suite, so it's not obvious this specific file generation procedure adds any value.

What should we do with it?

### Component(s)

Archery, Integration, JavaScript

**Comment by trxcllnt** [comment]
We [use this argument](https://github.com/apache/arrow/blob/ad7f6ef3a97be6a3eb6fe37d0df1b1634db057d2/js/gulp/test-task.js#L118-L126) to generate the JSON files for debugging the integration tests in isolation. My Union PR updated the logic to just [call the function](https://github.com/apache/arrow/pull/37122/files#diff-c79bbf8ddeec028e14acfc7b1ebbf5378d5e2cb14e8d93e4d898fb421cbd243aR124-R125) directly (and updated the function to generate the current set of files).

It would be great if there was a way to generate these files without needing any of the Python dependencies installed or importing subsets of the integration logic. The only realistic way to debug a failure in the integration suite is to generate the JSON/C++/Java files and run the JS integration test binaries in the debugger. It is prohibitively clunky to connect a debugger to a node process launched by a Python process inside a docker container running the full integration suite.

**Comment by pitrou** [comment]
> It would be great if there was a way to generate these files without needing any of the Python dependencies installed or importing subsets of the integration logic. 

I don't think it is a desirable goal to make subsets of Archery dependency-free, or to write a standalone script that will represent its own maintenance hassle.

I understand the annoyance you're describing. Instead of having to generate the files at runtime, perhaps we could update https://github.com/apache/arrow-testing/tree/master/data/arrow-ipc-stream/integration with some files generated by a newer version of Arrow C++/PyArrow?
